### PR TITLE
PollUntilContextTimeout refactor error handling.

### PR DESCRIPTION
**Pull request by samiulsami**

### CODE REVIEW

File: james/types.go [hunk 1]
```diff
@@ -226,18 +226,22 @@ func (w *WebAdminClient) RenewSession() error {
 	return <-errChan
 }
 
-func (w *WebAdminClient) renew() error {
-	var lastErr error
-	err := wait.PollUntilContextTimeout(
+func (w *WebAdminClient) renew() (err error) {
```
Comment (Imtiaz246) [inline james/types.go]: No need to declare err in the return params. Just use the err returned from the wait.PollUntilContextTimeout() and return. Sometimes it confusing to track the err

File: james/types.go [hunk 2]
```diff
@@ -355,57 +356,53 @@ func (j *JMAPClient) RenewSession() error {
 	return <-errChan
 }
 
-func (j *JMAPClient) renew() error {
-	var lastErr error
-	err := wait.PollUntilContextTimeout(
+func (j *JMAPClient) renew() (err error) {
+	_ = wait.PollUntilContextTimeout(
```
Comment (Imtiaz246) [inline james/types.go]: No need to declare err in the return params. Just use the err returned from the wait.PollUntilContextTimeout() and return. Sometimes it confusing to track the err

File: james/types.go [hunk 3]
```diff
@@ -355,57 +356,53 @@ func (j *JMAPClient) RenewSession() error {
...
 		context.TODO(),
 		100*time.Millisecond,
 		5*time.Second,
 		true,
-		func(ctx context.Context) (bool, error) {
+		func(ctx context.Context) (done bool, pollErr error) {
```
Comment (Imtiaz246) [inline james/types.go]: pollErr -> err

File: james/types.go (hunk 1 above)
Comment (Imtiaz246) [inline james/types.go]: Create one renew function to fetch and create the clients.

**Review by Imtiaz246 — CHANGES_REQUESTED** [review]

File: james/types.go (hunk 1 above)
Comment (samiulsami) [reply james/types.go]: It was initially done that way, but the problem was that wait.PollUntilContextTimeout() only returns the "context deadline reached" error, but not the errors that caused it. The actual relevant errors were "lost" in the process.

Comment (samiulsami) [reply james/types.go]: This will take some major refactoring. Should probably do this in another PR.

**Review by Imtiaz246 — APPROVED** [review]

**Pull request merged**